os/drivers/bluetooth/virtual_ble/vble_handler.c

```diff
@@ -0,0 +1,472 @@
+/****************************************************************************
+ *
+ * Copyright 2022 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <sys/types.h>
+#include <tinyara/config.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+#include <debug.h>
+#include <time.h>
+#include <tinyara/lwnl/lwnl.h>
+#include <tinyara/net/if/ble.h>
+#include "vble_handler.h"
+
+ble_scan_whitelist g_scan_whitelist[SCAN_WHITELIST_SIZE] = {0, };
+trble_connected_list vble_connected_list;
+trble_scanned_device vble_scanned_items;
+trble_scan_filter g_filter;
+trble_device_connected vble_connect;
+trble_device_connected connected_device[TRBLE_MAX_CONNECTION_COUNT];
+trble_operation_handle vble_op_handle;
+static int g_vble_result = TRBLE_SUCCESS;
+static uint8_t data[] = {
+    0x02, 0x01, 0x05, 0x03, 0x19, 0x80, 0x01, 0x05, 0x03, 0x12, 0x18, 0x0f, 0x18
+};
+trble_data vble_op_data;
+vattr_table attr_table;
+
+static inline void vdev_reverse_mac(uint8_t *mac)
+{
+    int i;
+    int j;
+    uint8_t temp;
+
+    for (i = 0, j = TRBLE_BD_ADDR_MAX_LEN - 1; i < j; i++, j--) {
+        temp = mac[i];
+        mac[i] = mac[j];
+        mac[j] = temp;
+    }
+    return;
+}
+
+static void _whitelist_delete(trble_addr *addr)
+{
+    int i;
+    for (i = 0; i < SCAN_WHITELIST_SIZE; i++) {
+        if (memcmp(g_scan_whitelist[i].addr.mac, addr->mac, TRBLE_BD_ADDR_MAX_LEN) == 0) {
+            g_scan_whitelist[i].use = SCAN_WHITELIST_EMPTY;
+            memset(&g_scan_whitelist[i], 0, sizeof(ble_scan_whitelist));
+            return;
+        }
+    }
+}
+
+static void _whitelist_add(trble_addr *addr)
+{
+    int i;
+    for (i = 0; i < SCAN_WHITELIST_SIZE; i++) {
+        if (g_scan_whitelist[i].use != SCAN_WHITELIST_EMPTY) {
+            continue;
+        }
+        g_scan_whitelist[i].use = SCAN_WHITELIST_IN_USE;
+        memcpy(g_scan_whitelist[i].addr.mac, addr->mac, TRBLE_BD_ADDR_MAX_LEN);
+        return;
+    }
+}
+
+static void vble_callback_handler(lwnl_cb_status evt, int result)
+{
+    switch(evt.evt){
```

Comment:
Please add space after `switch` and before `{`.

Reply:
Done